libcloud/common/openstack.py

```diff
@@ -391,7 +391,8 @@ def parse_error(self):
             driver = self.connection.driver
             key_pair_name = context.get('key_pair_name', None)
 
-            if len(values) > 0 and values[0]['code'] == 404 and key_pair_name:
+            if len(values) > 0 and 'code' in values[0] and \
```

Comment:
I see that this is a fix for errors looking like this:
```TypeError: string indices must be integers```
The conditional is growing pretty large here, but it works fine :+1:

Reply:
Yes. I did not find a shorter way to do it.